path_selection/path_selection.go

```diff
@@ -0,0 +1,263 @@
+// path_selection.go - mixnet client path selection
+// Copyright (C) 2017  David Anthony Stainton
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package provider mixnet client path selection
+package path_selection
+
+import (
+	cryptorand "crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/katzenpost/core/crypto/ecdh"
+	"github.com/katzenpost/core/crypto/rand"
+	"github.com/katzenpost/core/epochtime"
+	"github.com/katzenpost/core/pki"
+	"github.com/katzenpost/core/sphinx"
+	"github.com/katzenpost/core/sphinx/commands"
+	"github.com/katzenpost/core/sphinx/constants"
+)
+
+// durationFromFloat returns millisecond time.Duration given a float64
+func durationFromFloat(delay float64) time.Duration {
+	return time.Duration(float64(delay) * float64(time.Millisecond))
```

Comment:
i think what you want is
```	return time.Duration(delay*float64(time.Nanosecond)) / time.Millisecond ```
also note the units that delay should be in.

Reply:
why?

Reply:
i guess as above works - you shouldn't need the float64() cast on delay though.  Just don't want to have to run it through a string parser.